### PR TITLE
Allow overriding environment variable scope

### PR DESCRIPTION
**Pull request by JeremySkinner**
Allows changing the environment variable scope from Process to User/Machine. Also updated `Stop-SshAgent` and `Get-SshAgent` to work with win32 openssh.

### CODE REVIEW

**Review by dahlbyk — COMMENTED** [review]
I assume this is still WIP, since `$Scope` isn't actually used yet.

Changes so far look great!

**Comment by JeremySkinner** [comment]
Yes, I did commit the follow up but looks like I forgot to push it. Will do so when I'm back at my pc.

File: src/Agent.ps1 [hunk 1]
```diff
@@ -1,58 +1,66 @@
...
+        $StartupType = "Manual",
+
+        [Parameter()]
+        [switch]
+        $Quiet,
+
+        [Parameter()]
+        [ValidateSet("Machine", "Process", "User")]
+        [string]
+        $Scope = "Process"
```
Comment (rkeithhill) [inline src/Agent.ps1]: Is this parameter being used?

Comment (JeremySkinner) [reply src/Agent.ps1]: It is now! :) I forgot to commit the change

**Comment by JeremySkinner** [comment]
@dahlbyk @rkeithhill I've pushed up the missing change.

This doesn't change the environment variable scope for the calls to `Get-TempEnv` in the main module file, I don't think this is a big deal, but do we want to introduce a global variable to control this? My inclination would be no

**Comment by dahlbyk** [comment]
`setenv` doesn't understand your third parameter:

https://github.com/dahlbyk/posh-sshell/blob/edcb790de086f245a3a00f5fac2fc3247a0cd134/src/Utils.ps1#L3-L6

> This doesn't change the environment variable scope for the calls to `Get-TempEnv` in the main module file, I don't think this is a big deal, but do we want to introduce a global variable to control this? My inclination would be no

`Get-TempEnv` is already explicitly [user-scoped](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettemppath?view=netframework-4.7.2#System_IO_Path_GetTempPath) rather than process-scoped, and it doesn't make sense for it to be machine-scoped, so I think this is fine. In fact, it doesn't seem useful or correct to allow `Start-SshAgent -Scope Machine` either, as this could leak the current user's credentials to another user. Am I missing a legitimate scenario for that?

**Comment by JeremySkinner** [comment]
@dahlbyk Good point, I'll remove the option for machine scope and leave just process/user. 

Not sure why my change to setenv isn't showing up in the PR, it shows in my local repo. I'll force push it up.

**Comment by JeremySkinner** [comment]
@dahlbyk force push seems to have done it, the changes are showing now and I've removed machine scope from the options list.

File: src/Utils.ps1 [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+        [string]
+        $value,
+
+        [Parameter()]
+        [ValidateSet("Process", "User")]
+        [string]
+        $Scope = "Process"
+    )
+
+    $type = [System.EnvironmentVariableTarget]::Process
```
Comment (dahlbyk) [inline src/Utils.ps1]: I'm reasonably certain you can just pass `$Scope` directly to `SetEnvironmentVariable()` and PS will automatically cast for you.

Comment (JeremySkinner) [reply src/Utils.ps1]: @dahlbyk  You're right. I've removed the conversion.

**Pull request merged**